### PR TITLE
[ARCTIC-1111] Address the deserializing exception of the array<string> type in the logstore

### PR DESCRIPTION
**Pull request by YesOrNo828**
## Why are the changes needed?

Address the deserializing exception of the array<string> type in the logstore

Fix: #1111 
## Brief change log

*(For example:)*
  - *Add LogRecordV1Test*
  - *The flink 1.12, 1.14, and 1.15 versions are affected.*
  - *Update the JsonToLogDataConverters class*

## How was this patch tested?
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / no) no 
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented) not

### CODE REVIEW

File: core/src/main/java/com/netease/arctic/log/JsonToLogDataConverters.java [hunk 1]
```diff
@@ -181,20 +181,19 @@ private JsonToLogDataConverter<T> createListConverter(Type type) {
...
       final ArrayNode node = (ArrayNode) jsonNode;
-      final Object[] array = (Object[]) Array.newInstance(elementClass, node.size());
+      Object[] array = null;
       for (int i = 0; i < node.size(); i++) {
         final JsonNode innerNode = node.get(i);
         Object value = elementConverter.convert(innerNode, context);
-        array[i] = convertSecondTimeIfNecessary(elementType, value);
+        Object flinkValue = convertSecondTimeIfNecessary(elementType, value);
+        if (array == null) {
+          Class<?> flinkValueClass = flinkValue.getClass();
```
Comment (zstraw) [inline core/src/main/java/com/netease/arctic/log/JsonToLogDataConverters.java:193]: Is it possible that flinkValue is null?

File: flink/v1.12/flink-format/src/main/java/com/netease/arctic/flink/shuffle/LogRecordV1.java [hunk 2]
```diff
@@ -353,20 +353,16 @@ public RowData createActualValue(Object[] objects, Type[] fieldTypes) {
...
-        obj = wrapIntoNullableConvert(type, obj);
+        obj = convertIfNecessary(type, obj);
         row.setField(i, obj);
       }
       return row;
     }
 
-    public Object wrapIntoNullableConvert(Type primitiveType, Object object) {
-      if (object == null) {
+    public Object convertIfNecessary(Type primitiveType, Object obj) {
```
Comment (YesOrNo828) [inline flink/v1.12/flink-format/src/main/java/com/netease/arctic/flink/shuffle/LogRecordV1.java:362]: This is LogData.Factory super method, so I move the nullable check into the `convertIfnecessary` method.

**Review by zstraw — APPROVED** [review]
LGTM

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**